### PR TITLE
fixing fast float stod conversion according to ST_Number schema specification

### PR DESCRIPTION
**Pull request by gangatp**
- Allowing leading + sign
- Not allowing **,** separated values 
     -   example:  for a comma separated value 123,456 . The result from from_chars will be result=123.0 and answer.ptr = ',456'
     
  auto answer = fast_float::from_chars(input.data(), input.data()+input.size(), result);
  if(answer.ec != std::errc()) {
    // check error
  }
  if(answer.ptr[0] == ',') {
    // unexpected delimiter
  }

### CODE REVIEW

File: Tests/CPP_Bindings/Source/Reader.cpp [hunk 1]
```diff
@@ -149,4 +149,10 @@ namespace Lib3MF
 		auto reader = model->QueryReader("3mf");
 		ASSERT_SPECIFIC_THROW(reader->ReadFromFile(sTestFilesPath + "/Reader/" + "GEN-M-ADA-ITEM-TRANSFORM-0.3mf"), ELib3MFException);
 	}
+
+
+	TEST_F(Reader, IntegrationTestError) {
```
Comment (martinweismann) [inline Tests/CPP_Bindings/Source/Reader.cpp:154]: good idea to add a test there.

Please explain what the content of the file is and why it should fail.
Also please change the name to something meaningfull.

File: Source/Common/NMR_StringUtils.cpp [hunk 2]
```diff
@@ -161,6 +161,10 @@ namespace NMR {
 		{
 			throw CNMRException(NMR_ERROR_EMPTYSTRINGTODOUBLECONVERSION);
 		}
+		else if (answer.ptr && answer.ptr[0] == ',') // Invalidate comma as decimal separator
+		{
+			throw CNMRException(NMR_ERROR_INVALIDSTRINGTODOUBLECONVERSION);
```
Comment (martinweismann) [inline Source/Common/NMR_StringUtils.cpp:166]: let's also add a test for that, please.

Comment (alexanderoster) [reply Source/Common/NMR_StringUtils.cpp:166]: I would prefer that we do never add such style of code. While it is shorter, it is a landmine and who knows on what situation and platform (or little change by somebody) answer.ptr[0] is evaluated when answer.ptr == nullptr.

Making two if statements out of it is much easier to read and much safer.

Comment (gangatp) [reply Source/Common/NMR_StringUtils.cpp:166]: if(answer.ptr){
    if(answer.ptr[0] == ','){
        throw here
    }
}

I can make this into two if statements for readability, but that would mean the same. Did you mean anything else?

Comment (alexanderoster) [reply Source/Common/NMR_StringUtils.cpp:166]: That's what I meant, yes :)

**Review by martinweismann — APPROVED** [review]

**Pull request merged**